experimental/saiexperimentaldashha.h

```diff
@@ -0,0 +1,442 @@
+
+    /**
+     * @brief Number of HA heartbeats missed before declaring Peer unreachable
+     *
+     * @type sai_uint32_t
+     * @flags CREATE_AND_SET
+     * @default 0
+     */
+    SAI_DASH_HA_ATTR_HEARTBEAT_COUNT,
+
+    /**
+     * @brief Time in ms required for network to converge after switchover
+     *
+     * @par Any traffic directed to wrong DPU after this time will be blackholed
+     *
+     * @type sai_uint32_t
+     * @flags CREATE_AND_SET
+     * @default 100
+     */
+    SAI_DASH_HA_ATTR_VIP_CONVERGENCE_TIMEOUT,
+
+    /**
+     * @brief List of HA VIP status
+     *
+     * @type sai_dash_ha_vip_status_list_t
+     * @flags READ_ONLY
+     */
+    SAI_DASH_HA_ATTR_VIP_STATUS_LIST,
+
+    /**
+     * @brief End of attributes
+     */
+    SAI_DASH_HA_ATTR_END,
+
+    /** Custom range base value */
+    SAI_DASH_HA_ATTR_CUSTOM_RANGE_START = 0x10000000,
+
+    /** End of custom range base */
+    SAI_DASH_HA_ATTR_CUSTOM_RANGE_END
+
+} sai_dash_ha_attr_t;
+
+/**
+ * @brief Key for DASH HA FSM APIs
+ */
+typedef struct _sai_dash_ha_fsm_entry_t
+{
+    /**
+     * @brief Switch ID
+     *
+     * @objects SAI_OBJECT_TYPE_DASH_HA
+     */
+    sai_object_id_t dash_ha_id;
+
+    /**
+     * @brief VIP ID to which this API applies
+     *
+     * @par If VIP ID is not specified then this applies to all VIP IDs
+     * on the DPU
+     */
+    uint32_t vip_id;
+} sai_dash_ha_fsm_entry_t;
+
+/**
+ * @brief Attribute ID for DASH HA FSM API
+ */
+typedef enum _sai_dash_ha_fsm_attr_t
+{
+    /**
+     * @brief Start of attributes
+     */
+    SAI_DASH_HA_FSM_ATTR_START,
+
+    /**
+     * @brief Force flag to ignore optimizations
+     *
+     * @type bool
+     * @flags CREATE_ONLY
+     * @default false
+     */
+    SAI_DASH_HA_FSM_ATTR_FORCE = SAI_DASH_HA_FSM_ATTR_START,
+
+    /**
+     * @brief End of attributes
+     */
+    SAI_DASH_HA_FSM_ATTR_END,
+
+    /** Custom range base value */
+    SAI_DASH_HA_FSM_ATTR_CUSTOM_RANGE_START = 0x10000000,
+
+    /** End of custom range base */
+    SAI_DASH_HA_FSM_ATTR_CUSTOM_RANGE_END
+
+} sai_dash_ha_fsm_attr_t;
+
+/**
+ * @brief Counter IDs in sai_get_dash_ha_stats() call
+ */
+typedef enum _sai_dash_ha_stat_t
+{
+    /** DASH HA stat heartbeat messages sent */
+    SAI_DASH_HA_STAT_HEARTBEAT_MSG_SENT,
+
+    /** DASH HA stat heartbeat messages received */
+    SAI_DASH_HA_STAT_HEARTBEAT_MSG_RECEIVED,
+
+    /** DASH HA stat flow sync requests sent */
+    SAI_DASH_HA_STAT_FLOW_SYNC_REQ_SENT,
+
+    /** DASH HA stat flow sync acknowledgements sent */
+    SAI_DASH_HA_STAT_FLOW_SYNC_ACK_RECEIVED,
+
+} sai_dash_ha_stat_t;
+
+/**
+ * @brief DASH HA event type
+ */
+typedef enum _sai_dash_ha_fsm_event_t
+{
+    /** HA Flow reconcile API needs to be invoked */
+    SAI_DASH_HA_FSM_EVENT_FLOW_RECONCILE_REQUIRED,
+
+    /** HA Stop followed by HA Start API needs to be invoked */
+    SAI_DASH_HA_FSM_EVENT_STOP_START_REQUIRED,
+
+} sai_dash_ha_fsm_event_t;
```

Comment:
this seems to be a missing feature:

for every HA state change, we need it to be reported all the way back to SDN controller, and this event is missing here. this event is critical for debugging and monitoring, so it is better to add it.